api/src/main/java/org/openmrs/module/fgh/mpi/MpiHttpClient.java

```diff
@@ -146,20 +283,44 @@ public void submitPatient(String patientData) throws Exception {
 	 * @throws Exception
 	 */
 	private <T> T submitRequest(String requestPath, String data, Class<T> responseType) throws Exception {
-		initIfNecessary();
+		AdministrationService adminService = Context.getAdministrationService();
+		
+		serverBaseUrl = adminService.getGlobalProperty(GP_MPI_BASE_URL);
 		String url = serverBaseUrl + "/" + requestPath;
-		HttpsURLConnection connection = (HttpsURLConnection) new URL(url).openConnection();
+		
+		String contentType = adminService.getGlobalProperty(GP_MPI_APP_CONTENT_TYPE);
+		String authenticationType = adminService.getGlobalProperty(GP_AUTHENTICATION_TYPE);
+		
+		HttpURLConnection connection = null;
+		
+		if (authenticationType.equals(AUTHENTICATION_OUTH_TYPE)) {
+			retriveAccessToken(responseType);
+			
+			connection = (HttpURLConnection) new URL(url).openConnection();
+			
+			String authHeaderValue = "bearer " + this.tokenInfo.getAccess_token();
+			
+			connection.setRequestProperty("Authorization", authHeaderValue);
+		} else if (authenticationType.equals(AUTHENTICATION_SSL_TYPE)) {
```

Comment:
SSL is not an authentication type, for more clarity you can name it may be something like AUTHENTICATION_TYPE_CLIENT_CERTIFICATE, technically speaking OpenCR has no authentication type it is using, it merely depends on mTLS between client and server.

Reply:
Do you mind making authentication type an enum to have more strictness?

Reply:
Done